snapcraft/internal/errors.py

```diff
@@ -25,6 +25,16 @@
     'minLength': 'minimum length is {validator_value}',
 }
 
+# dict with nice error messages for invalid names in parts, hooks and apps
+_PROPERTY_ERROR_MESSAGES = {
+    "parts": "Part name does not match "
+    "'^(?!plugins$)[a-z0-9][a-z0-9+-\\/]*$'",
+    "apps": "App name does not match "
+    "'^[a-zA-Z0-9](?:-?[a-zA-Z0-9])*$'",
+    "hooks": "Hook name does not match "
+    "'^[a-zA-Z0-9](?:-?[a-zA-Z0-9])*$'",
```

Comment:
Initially I created the error message by obtaining the regex from the schema itself then I decided to change it, but I do consider you're right and the best approach would be to have the validation error messages in the schema, I'm going to do some more research to find a better solution thanks